cmd/util/cmd/epochs/cmd/recover.go

```diff
@@ -83,103 +72,142 @@ func generateRecoverEpochTxArgs(cmd *cobra.Command, args []string) {
 		log.Fatal().Err(err).Msg("failed")
 	}
 
-	// extract arguments from recover epoch tx from snapshot
-	txArgs := extractRecoverEpochArgs(snapshot)
+	return snapshot
+}
 
-	// encode to JSON
-	encodedTxArgs, err := epochcmdutil.EncodeArgs(txArgs)
-	if err != nil {
-		log.Fatal().Err(err).Msg("could not encode recover epoch transaction arguments")
-	}
+// generateRecoverEpochTxArgs generates recover epoch transaction arguments from a root protocol state snapshot and writes it to a JSON file
+func generateRecoverEpochTxArgs(getSnapshot func() *inmem.Snapshot) func(cmd *cobra.Command, args []string) {
+	return func(cmd *cobra.Command, args []string) {
+		stdout := cmd.OutOrStdout()
 
-	// write JSON args to stdout
-	_, err = stdout.Write(encodedTxArgs)
-	if err != nil {
-		log.Fatal().Err(err).Msg("could not write jsoncdc encoded arguments")
+		// extract arguments from recover epoch tx from snapshot
+		txArgs := extractRecoverEpochArgs(getSnapshot())
+
+		// encode to JSON
+		encodedTxArgs, err := epochcmdutil.EncodeArgs(txArgs)
+		if err != nil {
+			log.Fatal().Err(err).Msg("could not encode recover epoch transaction arguments")
+		}
+
+		// write JSON args to stdout
+		_, err = stdout.Write(encodedTxArgs)
+		if err != nil {
+			log.Fatal().Err(err).Msg("could not write jsoncdc encoded arguments")
+		}
 	}
 }
 
 // extractResetEpochArgs extracts the required transaction arguments for the `resetEpoch` transaction
 func extractRecoverEpochArgs(snapshot *inmem.Snapshot) []cadence.Value {
 	epoch := snapshot.Epochs().Current()
-	ids, err := epoch.InitialIdentities()
+
+	ids, err := snapshot.Identities(filter.IsValidProtocolParticipant)
```

Comment:
```suggestion
	currentEpochIdentities, err := snapshot.Identities(filter.IsValidProtocolParticipant)
```
Suggesting a more descriptive name since this is a fairly long function.